### PR TITLE
backport: bitcoin/bitcoin#22229 test: consolidate to f-strings and related fixes

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
Just one backport bitcoin#22229 because it is a big size, even though there's nothing non-trivial in it.

Though, even it is called as part I, there has not been part II yet.

## What was done?
Some preparation, code unifications to make bitcoin#22229 with less conflicts and finally backport of itselfl.

## How Has This Been Tested?
Run unit & functional test

## Breaking Changes
N/A

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by knst** [comment]
> This pull request has conflicts, please rebase.

That's a lie!

**Review by PastaPastaPasta — APPROVED** [review]
utACK fc8436832b9b230e3aab2592245c6f71b618f3f6

File: test/functional/feature_notifications.py [hunk 1]
```diff
@@ -65,6 +66,35 @@ def run_test(self):
         for tx_file in os.listdir(self.walletnotify_dir):
             os.remove(os.path.join(self.walletnotify_dir, tx_file))
 
+        if self.is_wallet_compiled():
+            # Setup the descriptors to be imported to the wallet
+            seed = "cTdGmKFWpbvpKQ7ejrdzqYT2hhjyb3GPHnLAK7wdi5Em67YLwSm9"
```
Comment (UdjinM6) [inline test/functional/feature_notifications.py:71]: cafc78c0869c1b6712829b8944dfd94b753d07ba : linter complains `test/functional/feature_notifications.py:71:13: F841 local variable 'seed' is assigned to but never used`

File: test/functional/feature_loadblock.py [hunk 2]
```diff
@@ -45,17 +45,17 @@ def run_test(self):
...
-            cfg.write("output_file={}\n".format(bootstrap_file))
-            cfg.write("max_height=100\n")
+            cfg.write(f"datadir={data_dir}\n")
+            cfg.write(f"rpcuser={node_url.username}\n")
+            cfg.write(f"rpcpassword={node_url.password}\n")
+            cfg.write(f"port={node_url.port}\n")
+            cfg.write(f"host={node_url.hostname}\n")
+            cfg.write(f"output_file={bootstrap_file}\n")
+            cfg.write(f"max_height=100\n")
             cfg.write("netmagic=fcc1b7dc\n")
```
Comment (UdjinM6) [inline test/functional/feature_loadblock.py:55]: 22229: missed `f` here (to align with upstream even though `f` is not really needed here)

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — DISMISSED** [review]
utACK a15e33937c8551acd73e8d992824b7cc9659ab8a

**Review by PastaPastaPasta — DISMISSED** [review]
utACK a15e339, good to merge once CI is happy

**Comment by knst** [comment]
CI fails for sqlite build

**Comment by UdjinM6** [comment]
reverting/dropping 1b6baa5899 fixes it https://gitlab.com/UdjinM6/dash/-/pipelines/1511972465

**Comment by knst** [comment]
> reverting/dropping https://github.com/dashpay/dash/commit/1b6baa58994ad596c7d8da1cf193146d6fecd4f2 fixes it https://gitlab.com/UdjinM6/dash/-/pipelines/1511972465


yes I am looking to the same commit, but not sure how exactly to fix it yet. I will try to find a fix it and if it is not easy I will just drop it.

**Comment by knst** [comment]
I dropped that problematic commit for feature_notifications.py; will resolve issue some other day

**Review by PastaPastaPasta — APPROVED** [review]
utACK https://github.com/dashpay/dash/commit/7c6c93d201b7338257ebc413f58b22bea490b8a2

**Pull request merged**